pkg/mcp/server/server.go

```diff
@@ -81,108 +73,29 @@ type AuthChecker interface {
 	Check(authInfo credentials.AuthInfo) error
 }
 
-type newPushResponseState int
-
-const (
-	newPushResponseStateReady newPushResponseState = iota
-	newPushResponseStateClosed
-)
-
 // watch maintains local push state of the most recent watch per-type.
 type watch struct {
 	// only accessed from connection goroutine
 	cancel          func()
 	nonce           string // most recent nonce
 	nonceVersionMap map[string]string
 
-	// NOTE: do not hold `mu` when reading/writing to this channel.
-	newPushResponseReadyChan chan newPushResponseState
-
 	mu                      sync.Mutex
-	newPushResponse         *source.WatchResponse
 	mostRecentNackedVersion string
-	timer                   *time.Timer
-	closed                  bool
-}
-
-func (w *watch) delayedPush() {
-	w.mu.Lock()
-	w.timer = nil
-	w.mu.Unlock()
-
-	select {
-	case w.newPushResponseReadyChan <- newPushResponseStateReady:
-	default:
-		time.AfterFunc(retryPushDelay, w.schedulePush)
-	}
-}
-
-// Try to schedule pushing a response to the node. The push may
-// be re-scheduled as needed. Additional care is taken to rate limit
-// re-pushing responses that were previously NACK'd. This avoid flooding
-// the node with responses while also allowing transient NACK'd responses
-// to be retried.
-func (w *watch) schedulePush() {
-	w.mu.Lock()
-
-	// close the watch
-	if w.closed {
-		// unlock before channel write
-		w.mu.Unlock()
-
-		select {
-		case w.newPushResponseReadyChan <- newPushResponseStateClosed:
-		default:
-			time.AfterFunc(retryPushDelay, w.schedulePush)
-		}
-		return
-	}
-
-	// no-op if the response has already be sent
-	if w.newPushResponse == nil {
-		w.mu.Unlock()
-		return
-	}
-
-	// delay re-sending a previously nack'd response
-	if w.newPushResponse.Version == w.mostRecentNackedVersion {
-		if w.timer == nil {
-			w.timer = time.AfterFunc(nackLimitFreq, w.delayedPush)
-		}
-		w.mu.Unlock()
-		return
-	}
-
-	// Otherwise, try to schedule the response to be sent.
-	if w.timer != nil {
-		if !w.timer.Stop() {
-			<-w.timer.C
-		}
-		w.timer = nil
-	}
-	// unlock before channel write
-	w.mu.Unlock()
-	select {
-	case w.newPushResponseReadyChan <- newPushResponseStateReady:
-	default:
-		time.AfterFunc(retryPushDelay, w.schedulePush)
-	}
 
+	queueResponse func(*source.WatchResponse) bool
```

Comment:
Minor nit: It is not clear what these are from first look. Comments here would help.